### PR TITLE
Deferrable mode for CreateBatchPredictionJobOperator

### PR DESCRIPTION
**Pull request by moiseenkov**
Deferrable mode for the `CreateBatchPredictionJobOperator` is introduced.

### CODE REVIEW

**Review by dabla — APPROVED** [review]
Nice refactoring @moiseenkov, async operators are better as those are non-blocking and don't hold up workers while waiting for a response.

You could improve the tests like I did with my async msgraph operator [here](https://github.com/infrabel/apache-airflow-providers-msgraph/blob/main/tests/unit/test_operator.py#L37), so that you simulate the whole async execution cycle including the trigger execution.  You'll see that I've create an [execute_operator](https://github.com/infrabel/apache-airflow-providers-msgraph/blob/main/tests/unit/base.py#L30) method, that way we simulate the whole behaviour in the same main thread and then we can do more than just simple verifications, we can actually also assert the result of that execution.  Maybe we could move that execute_operator helper method somewhere to some central helper method in the tests package so it can be re-used across different tests for async operators.  Doing so also allows you, like I did, to accumulate XCom interactions with the context.  Don't know what the other Airflow project maintainers think of this approach.

**Comment by VladaZakharova** [comment]
Hi @eladkal @potiuk !
Can we please merge these changes?

**Review by potiuk — APPROVED** [review]

**Pull request merged**
